gateway-js/src/schema-helper/addResolversToSchema.ts

```diff
@@ -19,7 +19,18 @@ export function addResolversToSchema(
 
     if (isAbstractType(type)) {
       for (const [fieldName, fieldConfig] of Object.entries(fieldConfigs)) {
-        if (fieldName.startsWith("__")) {
+        if (fieldName === "__resolveReference") {
+          type.extensions = {
+            ...type.extensions,
+            apollo: {
+              ...type.extensions.apollo,
+              subgraph: {
+                ...type.extensions.apollo?.subgraph,
+                resolveReference: fieldConfig,
+              }
+            },
+          };
+        } else if (fieldName.startsWith("__")) {
           (type as any)[fieldName.substring(2)] = fieldConfig;
```

Comment:
@trevor-scheer is this basically just for `resolveType` on abstract types and `isTypeOf` on object types? wonder if we should just special case them (which lets us drop the `type as any`).

Reply:
Also it looks like this entire directory (`gateway-js/src/schema-helper`, not to be confused with `subgraph-js/src/schema-helper`) is only used in tests? Can it be moved into `gateway-js/src/__tests__`, or maybe you can just use the version in `subgraph-js` directly from the tests?